src/main/java/org/spongepowered/common/data/processor/data/entity/BodyPartRotationalDataProcessor.java

```diff
@@ -0,0 +1,126 @@
+                Keys.RIGHT_LEG_ROTATION.getQuery()
+            )
+        ) {
+            return Optional.empty();
+        }
+        @SuppressWarnings("unchecked")
+        Map<BodyPart, Vector3d> bodyRotations = (Map<BodyPart, Vector3d>) container.getMap(Keys.BODY_ROTATIONS.getQuery()).get();
+        Vector3d headRotation = (Vector3d) container.get(Keys.HEAD_ROTATION.getQuery()).get();
+        Vector3d chestRotation = (Vector3d) container.get(Keys.CHEST_ROTATION.getQuery()).get();
+        Vector3d leftArmRotation = (Vector3d) container.get(Keys.LEFT_ARM_ROTATION.getQuery()).get();
+        Vector3d rightArmRotation = (Vector3d) container.get(Keys.RIGHT_ARM_ROTATION.getQuery()).get();
+        Vector3d leftLegRotation = (Vector3d) container.get(Keys.LEFT_LEG_ROTATION.getQuery()).get();
+        Vector3d rightLegRotation = (Vector3d) container.get(Keys.RIGHT_LEG_ROTATION.getQuery()).get();
+        
+        data.set(Keys.BODY_ROTATIONS, bodyRotations);
+        data.set(Keys.HEAD_ROTATION, headRotation);
+        data.set(Keys.CHEST_ROTATION, chestRotation);
+        data.set(Keys.LEFT_ARM_ROTATION, leftArmRotation);
+        data.set(Keys.RIGHT_ARM_ROTATION, rightArmRotation);
+        data.set(Keys.LEFT_LEG_ROTATION, leftLegRotation);
+        data.set(Keys.RIGHT_LEG_ROTATION, rightLegRotation);
+        return Optional.of(data);
+    }
+
+    @Override
+    public DataTransactionResult remove(DataHolder dataHolder) {
+        return DataTransactionResult.failNoData();
+    }
+
+    @Override
+    protected boolean doesDataExist(EntityArmorStand dataHolder) {
+        return true;
+    }
+
+    @Override
+    protected boolean set(EntityArmorStand dataHolder, Map<Key<?>, Object> keyValues) {
+        dataHolder.setHeadRotation(VecHelper.toRotation((Vector3d) keyValues.get(Keys.HEAD_ROTATION)));
+        dataHolder.setBodyRotation(VecHelper.toRotation((Vector3d) keyValues.get(Keys.CHEST_ROTATION)));
+        dataHolder.setLeftArmRotation(VecHelper.toRotation((Vector3d) keyValues.get(Keys.LEFT_ARM_ROTATION)));
+        dataHolder.setRightArmRotation(VecHelper.toRotation((Vector3d) keyValues.get(Keys.RIGHT_ARM_ROTATION)));
+        dataHolder.setLeftLegRotation(VecHelper.toRotation((Vector3d) keyValues.get(Keys.LEFT_LEG_ROTATION)));
+        dataHolder.setRightLegRotation(VecHelper.toRotation((Vector3d) keyValues.get(Keys.RIGHT_LEG_ROTATION)));
+        return true;
+    }
+
+    @Override
+    protected Map<Key<?>, ?> getValues(EntityArmorStand dataHolder) {
+        Map<Key<?>, Object> values = Maps.newHashMapWithExpectedSize(6);
+        DataWatcher dataWatcher = dataHolder.getDataWatcher();
+        
+        values.put(Keys.HEAD_ROTATION, VecHelper.toVector(dataHolder.getHeadRotation()));
+        values.put(Keys.CHEST_ROTATION, VecHelper.toVector(dataHolder.getBodyRotation()));
+        values.put(Keys.LEFT_ARM_ROTATION, VecHelper.toVector(dataWatcher.getWatchableObjectRotations(13)));
```

Comment:
Why use the internal data watcher? there's already a field available, `leftArmRotation`